From d6c64114b756b673485f70911c02f97e7f515237 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Wed, 27 Nov 2024 12:19:14 +0100
Subject: [PATCH] Increase the default test timeout

For some reason, linking parts of the test programs with GCC 13 is much
slower than it is with GCC 14. Workaround that by just doubling the test
timeout for now.
---
 testsuite/driver/runtests.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/testsuite/driver/runtests.py b/testsuite/driver/runtests.py
index 151b0932e6..ec79ca13ee 100644
--- a/testsuite/driver/runtests.py
+++ b/testsuite/driver/runtests.py
@@ -315,6 +315,9 @@ testopts_ctx_var.set(TestOptions())
 if config.timeout == -1:
     config.timeout = int(read_no_crs(config.top / 'timeout' / 'calibrate.out'))
 
+# XXX: Without this, some tests timeout on the Alpine CI.
+config.timeout *= 2
+
 print('Timeout is ' + str(config.timeout))
 print('Known ways: ' + ', '.join(config.other_ways))
 print('Run ways: ' + ', '.join(config.run_ways))
